File: ruby/scrabble-score/scrabble.rb
class Scrabble

  LETTER_VALUES = {
    A: 1, B: 3, C: 3, D: 2, E: 1, F: 4, G: 2, H: 4,
    I: 1, J: 8, K: 5, L: 1, M: 3, N: 1, O: 1, P: 3,
    Q:10, R: 1, S: 1, T: 1, U: 1, V: 4, W: 4, X: 8,
    Y: 4, Z:10
  }

  attr_reader :word
  def initialize(word)
    @word = (word || "").gsub(/\s+/, "").upcase
  end

  def score
    points = 0
    word.each_char {|char| points += LETTER_VALUES[char.to_sym]}
    points
  end

  def self.score(word)
    self.new(word).score
  end

end

File: python/scrabble-score/scrabble.py
class Scrabble:

    LETTER_VALUES = {
        'a': 1, 'b': 3, 'c': 3, 'd': 2, 'e': 1, 'f': 4, 'g': 2, 'h': 4,
        'i': 1, 'j': 8, 'k': 5, 'l': 1, 'm': 3, 'n': 1, 'o': 1, 'p': 3,
        'q': 10, 'r': 1, 's': 1, 't': 1, 'u': 1, 'v': 4, 'w': 4, 'x': 8,
        'y': 4, 'z': 10
    }

    @classmethod
    def score(cls, word):
        if not cls.valid(word):
            return 0
        return sum([cls.LETTER_VALUES[c.lower()] for c in word])

    @classmethod
    def valid(cls, word):
        return word.isalpha()


def score(word):
    return Scrabble.score(word)
